internal/serviceconfig/serviceconfig_test.go

```diff
@@ -133,3 +135,57 @@ func TestBalancerConfigUnmarshalJSON(t *testing.T) {
 		})
 	}
 }
+
+func TestBalancerConfigMarshalJSON(t *testing.T) {
+	tests := []struct {
+		name     string
+		bc       BalancerConfig
+		wantJSON string
+	}{
+		{
+			name: "OK",
+			bc: BalancerConfig{
+				Name:   testBalancerBuilderName,
+				Config: testBalancerConfig,
+			},
+			wantJSON: fmt.Sprintf("[{%q: %v}]", testBalancerBuilderName, testBalancerConfigJSON),
+		},
+		{
+			name: "first balancer not registered",
+			bc: BalancerConfig{
+				Name:   testBalancerBuilderName,
+				Config: testBalancerConfig,
+			},
+			wantJSON: fmt.Sprintf("[{%q: %v}]", testBalancerBuilderName, testBalancerConfigJSON),
+		},
+		{
+			name: "balancer registered but builder not parser",
+			bc: BalancerConfig{
+				Name:   testBalancerBuilderNotParserName,
+				Config: nil,
+			},
+			wantJSON: fmt.Sprintf("[{%q: %v}]", testBalancerBuilderNotParserName, "{}"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			b, err := tt.bc.MarshalJSON()
+			if err != nil {
+				t.Fatalf("failed to marshal: %v", err)
+			}
+
+			str := string(b)
+			if str != tt.wantJSON {
```

Comment:
Combine the assignment and the conditional.

Reply:
Done